src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContractSet.cs

```diff
@@ -0,0 +1,528 @@
+                    AddXmlDataContract((XmlDataContract)dataContract);
+                }
+            }
+        }
+
+        void AddClassDataContract(ClassDataContract classDataContract)
+        {
+            if (classDataContract.BaseContract != null)
+            {
+                Add(classDataContract.BaseContract.StableName, classDataContract.BaseContract);
+            }
+            if (!classDataContract.IsISerializable)
+            {
+                if (classDataContract.Members != null)
+                {
+                    for (int i = 0; i < classDataContract.Members.Count; i++)
+                    {
+                        DataMember dataMember = classDataContract.Members[i];
+                        DataContract memberDataContract = GetMemberTypeDataContract(dataMember);
+                        //if (dataContractSurrogate != null && dataMember.MemberInfo != null)
+                        //{
+                        //    object customData = DataContractSurrogateCaller.GetCustomDataToExport(
+                        //                           dataContractSurrogate,
+                        //                           dataMember.MemberInfo,
+                        //                           memberDataContract.UnderlyingType);
+                        //    if (customData != null)
+                        //        SurrogateDataTable.Add(dataMember, customData);
+                        //}
+                        Add(memberDataContract.StableName, memberDataContract);
+                    }
+                }
+            }
+            AddKnownDataContracts(classDataContract.KnownDataContracts);
+        }
+
+        void AddCollectionDataContract(CollectionDataContract collectionDataContract)
+        {
+            if (collectionDataContract.IsDictionary)
+            {
+                ClassDataContract keyValueContract = collectionDataContract.ItemContract as ClassDataContract;
+                AddClassDataContract(keyValueContract);
+            }
+            else
+            {
+                DataContract itemContract = GetItemTypeDataContract(collectionDataContract);
+                if (itemContract != null)
+                    Add(itemContract.StableName, itemContract);
+            }
+            AddKnownDataContracts(collectionDataContract.KnownDataContracts);
+        }
+
+        void AddXmlDataContract(XmlDataContract xmlDataContract)
+        {
+            AddKnownDataContracts(xmlDataContract.KnownDataContracts);
+        }
+
+        void AddKnownDataContracts(DataContractDictionary knownDataContracts)
+        {
+            if (knownDataContracts != null)
+            {
+                foreach (DataContract knownDataContract in knownDataContracts.Values)
+                {
+                    Add(knownDataContract);
+                }
+            }
+        }
+
+        internal XmlQualifiedName GetStableName(Type clrType)
+        {
+            //if (dataContractSurrogate != null)
```

Comment:
I'd prefer to keep the code for surrogate in-place for now. We probably would need to provide some alternative for surrogate later in the future. Keeping the code in-place would make the future work easier. Or I can put the code in #if SUPPORT_SURROGATE, would that be better? 

---
In reply to: [82293921](https://github.com/dotnet/corefx/pull/12378#discussion_r82293921) [](ancestors = 82293921)